tei2slob/__init__.py

```diff
@@ -157,16 +157,23 @@ def mk_html_element(self, element, html_parent):
                 ol.attrib['class'] += ' single'
 
     def __iter__(self):
-        for _, element in etree.iterparse(self.input_file):
-
-            if element.tag == TAG_HEADER:
-                yield from self._parse_header(element)
-                element.clear()
-
-            if element.tag == TAG_ENTRY:
-                yield from self._parse_entry(element)
-                element.clear()
-
+        input_files = [self.input_file]
+        while input_files:
+            for _, element in etree.iterparse(input_files.pop()):
+
+                if element.tag == TAG_HEADER:
+                    yield from self._parse_header(element)
+                    element.clear()
+
+                if element.tag == TAG_ENTRY:
+                    yield from self._parse_entry(element)
+                    element.clear()
+
+                if element.tag == '{http://www.w3.org/2001/XInclude}include':
+                    include_file = os.path.join(
+                            os.path.dirname(self.input_file),
+                            element.attrib['href'])
+                    input_files.append(include_file)
```

Comment:
A more robust implementation should check that the file to be included hasn't been seen already (and that it exists), and either ignore them or bail, so that the program doesn't get into  infinite loop on a bad dictionary. But if you want to keep it simple and assume input dictionaries are reasonable I guess that's ok too...